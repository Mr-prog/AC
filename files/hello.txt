  section_data
  hello $ hello
  world $ , world!
  section_program
  start:
    LOADMEMORY hello R6
    LOADMEMORY world R7
    WRITE_PORT R6
    WRITE_PORT R7
    NOP