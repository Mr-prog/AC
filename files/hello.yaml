; hello_user_name.asm
; Запросить у пользователя имя и вывести приветствие

start:
    LOADI "What is your name?", R1  ; Загрузить строку запроса в регистр
    WRITE_PORT R1, POUT             ; Вывести строку запроса на экран

    READ_PORT PIN, R2               ; Считать имя пользователя с порта ввода
    LOADI "Hello, ", R3             ; Загрузить строку приветствия
    WRITE_PORT R3, POUT             ; Вывести строку "Hello, " на экран
    WRITE_PORT R2, POUT             ; Вывести имя пользователя на экран
    LOADI "!", R4                   ; Загрузить символ восклицательного знака
    WRITE_PORT R4, POUT             ; Вывести восклицательный знак

end:
    NOP                             ; Завершение программы
