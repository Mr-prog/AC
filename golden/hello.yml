in_source: |
  section_data
  hello $ hello
  world $ , world!
  section_program
  start:
    LOADMEMORY hello R6
    LOADMEMORY world R7
    WRITE_PORT R6
    WRITE_PORT R7
    NOP
in_stdin: ''
out_code: |
  Instruction counter: 0, LOADMEMORY ['hello', 'R6']
  Instruction counter: 1, LOADMEMORY ['world', 'R7']
  Instruction counter: 2, WRITE_PORT ['R6']
  Instruction counter: 3, WRITE_PORT ['R7']
  Instruction counter: 4, NOP []
out_stdout: 'hello, world!'
out_log: |
  DEBUG    root:machine.py:233 LOADMEMORY ['hello', 'R6'] {PC: 0, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: , REG7: , REG8: , REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: []}
  DEBUG    root:machine.py:233 LOADMEMORY ['world', 'R7'] {PC: 1, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: hello, REG7: , REG8: , REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: []}
  DEBUG    root:machine.py:233 WRITE_PORT ['R6'] {PC: 2, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: hello, REG7: , world!, REG8: , REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: []}
  DEBUG    root:machine.py:233 WRITE_PORT ['R7'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: hello, REG7: , world!, REG8: , REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['h', 'e', 'l', 'l', 'o']}
  DEBUG    root:machine.py:233 NOP [] {PC: 4, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: hello, REG7: , world!, REG8: , REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['h', 'e', 'l', 'l', 'o', ',', ' ', 'w', 'o', 'r', 'l', 'd', '!']}
